solr/core/src/test/org/apache/solr/metrics/JvmMetricsTest.java

```diff
@@ -92,9 +91,10 @@ public void testSystemProperties() {
     }
     SolrMetricManager metricManager = jetty.getCoreContainer().getMetricManager();
     Map<String, Metric> metrics = metricManager.registry("solr.jvm").getMetrics();
-    MetricsMap map =
-        (MetricsMap) ((SolrMetricManager.GaugeWrapper) metrics.get("system.properties")).getGauge();
-    assertNotNull(map);
+    Metric metric = metrics.get("system.properties");
+    assertNotNull(metrics.toString(), metric);
+    MetricsMap map = (MetricsMap) ((SolrMetricManager.GaugeWrapper<?>) metric).getGauge();
+    assertNotNull(metrics.toString(), map);
```

Comment:
This addresses the NPE by adding an assertion. Doesn't fix the underlying cause - haven't found one yet.